tests/builder/subcommands.rs

```diff
@@ -630,3 +631,86 @@ fn duplicate_subcommand_alias() {
         .subcommand(Command::new("unique").alias("repeat"))
         .build();
 }
+
+#[test]
+fn subcommand_help_header() {
+    static VISIBLE_ALIAS_HELP: &str = "\
+Usage: clap-test [COMMAND]
+
+Commands:
+  help  Print this message or the help of the given subcommand(s)
```

Comment:
Before this can move forward, we need to come to a resolution on `help`.  The initial discussion should be on #1553.  If we need to make new issues off of that, then we can migrate from there.